src/lib/clk.c

```diff
@@ -115,6 +115,18 @@ uint64_t clock_ms_to_ticks(int clock, uint64_t ms)
 	return ticks;
 }
 
+uint64_t clock_ticks_per_sample(int clock, uint32_t sample_rate)
+{
+	struct clock_info *clk_info = clocks_get() + clock;
+	uint32_t ticks_per_msec =
+		clk_info->freqs[clk_info->current_freq_idx].ticks_per_msec;
+	uint64_t ticks_per_sample = sample_rate ? ticks_per_msec * 1000 / sample_rate : 0;
+
+	platform_shared_commit(clk_info, sizeof(*clk_info));
```

Comment:
Btw, why are you committing this ? You are not changing anything in clk. I would have expected to invalidate this before we calculate the ticks ?

Reply:
> Btw, why are you committing this ? You are not changing anything in clk. I would have expected to invalidate this before we calculate the ticks ?

Indeed, platform_shared_get() should be used to invalidate the clk_info. It was copy-paste from clock_ms_to_ticks(), let me correct it.